commit c4cb247732767aed76502069d0b3040c4c4e5123
Author: Nils Philippsen <nils@redhat.com>
Date:   Fri Jul 31 16:21:53 2009 +0200

    patch: array-out-of-bounds
    
    Squashed commit of the following:
    
    commit 337bcefaa7a67931095b74317a266a1244978ab6
    Author: Nils Philippsen <nils@redhat.com>
    Date:   Fri Jul 31 16:03:28 2009 +0200
    
        fix array subscript out of bounds errors (#133121)

diff --git a/src/xcam.c b/src/xcam.c
index 2d494a5..f6859b7 100644
--- a/src/xcam.c
+++ b/src/xcam.c
@@ -1289,7 +1289,7 @@ save_frame_button (GtkWidget * widget, gpointer client_data,
 
       /* test for pnm formats */
       strncpy (testfilename, preferences.filename, sizeof (testfilename));
-      testfilename[sizeof (testfilename)] = 0;
+      testfilename[sizeof (testfilename) - 1] = 0;
       g_strreverse (testfilename);
       if (!((!strncmp (testfilename, "mnp.", 4)) ||
 	    (!strncmp (testfilename, "mgp.", 4)) ||
diff --git a/src/xscanimage.c b/src/xscanimage.c
index a36324f..065923d 100644
--- a/src/xscanimage.c
+++ b/src/xscanimage.c
@@ -1284,7 +1284,7 @@ scan_dialog (GtkWidget * widget, gpointer call_data)
     {				/* We are running in standalone mode */
       /* test for pnm formats */
       strncpy (testfilename, preferences.filename, sizeof (testfilename));
-      testfilename[sizeof (testfilename)] = 0;
+      testfilename[sizeof (testfilename) - 1] = 0;
       g_strreverse (testfilename);
       if (!((!strncmp (testfilename, "mnp.", 4)) ||
 	    (!strncmp (testfilename, "mgp.", 4)) ||
